llvm/lib/Target/AMDGPU/SIRegisterInfo.td

```diff
@@ -492,6 +493,13 @@ def VGPR_32 : RegisterClass<"AMDGPU", !listconcat(Reg32Types.types, Reg16Types.t
   let Weight = 1;
 }
 
+def VGPR_32_all : RegisterClass<"AMDGPU", [vtAny32], 32,
```

Comment:
Definitely should not be adding a new register class. Adding vtAny32 to the existing 32-bit classes may be the right solution, but I was originally thinking this would not be necessary